src/globus_sdk/services/timer/data.py

```diff
@@ -123,8 +123,26 @@ def from_transfer_data(
         transfer_action_url = slash_join(
             get_service_url("actions", environment=environment), "transfer/transfer/run"
         )
-        # dict will either convert a `TransferData` object or leave us with a dict here
-        callback_body = {"body": dict(transfer_data)}
+        body = dict(transfer_data)
+        # Use `"DATA" in body` as a proxy to check, if `transfer_data` is a
+        # dict, whether it's just dictified TransferData (which we still need
+        # to convert) or actually the correct data to send to Transfer AP
+        if isinstance(transfer_data, TransferData) or "DATA" in body:
```

Comment:
`body.get("DATA_TYPE") == "transfer"` would be a slightly less hacky way of doing this.